src/libstd/error.rs

```diff
@@ -221,6 +222,24 @@ impl Error for ! {
     fn description(&self) -> &str { *self }
 }
 
+#[unstable(feature = "allocator_api",
+           reason = "the precise API and guarantees it provides may be tweaked.",
+           issue = "27700")]
+impl Error for allocator::AllocErr {
```

Comment:
nit: this would read better (and be more consistent with e.g. `io::Error`) if this was `allocator::Error`